### PR TITLE
Add type declaration file for v0.7.0

### LINKED ISSUE #2841: Typescript Type Definitions in DefinitelyTyped
# Feature Request

- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Mac OSX 10.15
- **TensorFlow installed from (our builds, or upstream TensorFlow)**: Upstream
- **TensorFlow version (use command below)**: 1.14
- **Python version**: 3.7
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

I have added types for the DeepSpeech module v0.6.0 to [`DefinitelyTyped`](https://github.com/DefinitelyTyped/DefinitelyTyped).
If I could have a maintainer review my PR there to verify I covered all the Node.js bindings that would be appreciated (DefinitelyTyped/DefinitelyTyped#43321).
If the maintainers prefer a PR against this repo with the types, that is also something I am open to doing.

### PR DESCRIPTION
**Pull request by piraka9011**
Add a type declaration file for the latest DeepSpeech version.
Update documentation on how to generate and `index.d.ts`.

Closes #2841

Signed-off-by: Anas Abou Allaban <aabouallaban@pm.me>

### CODE REVIEW

File: native_client/javascript/README.md [hunk 1]
```diff
@@ -1 +1,13 @@
 Full project description and documentation on GitHub: [https://github.com/mozilla/DeepSpeech](https://github.com/mozilla/DeepSpeech).
+
+## Generating TypeScript Type Definitions
```
Comment (lissyx) [inline native_client/javascript/README.md:3]: Could you add doc on how to:
 - use typescript'd deepspeech ?
 - run tests

File: native_client/javascript/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "esModuleInterop": true,
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true
+    },
+    "files": [
+        "index.d.ts",
```
Comment (lissyx) [inline native_client/javascript/tsconfig.json:16]: this is broken, you can't have `,` here

File: native_client/javascript/README.md [hunk 3]
```diff
@@ -1 +1,29 @@
...
+
+```sh
+npm install -g dts-gen
+dts-gen --module deepspeech --file index.d.ts
+```
+
+### Example usage
+
+```javascript
+/// index.ts
```
Comment (lissyx) [inline native_client/javascript/README.md:18]: Should we write a `index.ts` next to our current `index.js` ? How do we package that and run it as typescript ?

File: native_client/javascript/README.md [hunk 4]
```diff
@@ -1 +1,29 @@
...
+import DeepSpeech from "deepspeech";
+
+let modelPath = '/path/to/model';
+let model = new DeepSpeech.Model(modelPath);
+model.setBeamWidth(1000);
+
+let audioBuffer = ...;
```
Comment (lissyx) [inline native_client/javascript/README.md:25]: `[]` instead of `...`

File: native_client/javascript/README.md [hunk 5]
```diff
@@ -1 +1,29 @@
...
+model.sttWithMetadata(audioBuffer);
+
+FreeModel(model);
```
Comment (lissyx) [inline native_client/javascript/README.md:28]: `DeepSpeech.FreeModel(model)`

File: native_client/javascript/README.md (hunk 3 above)
Comment (lissyx) [reply native_client/javascript/README.md:18]: Ok, so we can use `tsc` after installing `typescript` NPM package.

Comment (lissyx) [reply native_client/javascript/README.md:18]: > Ok, so we can use `tsc` after installing `typescript` NPM package.

Hm that is just re-generating my `index.js`

File: native_client/javascript/README.md [hunk 6]
```diff
@@ -1 +1,29 @@
...
+## Generating TypeScript Type Definitions
+
+You can generate the TypeScript type declaration file using `dts-gen`.
+This requires a compiled/installed version of the DeepSpeech NodeJS client.
+
+To generate a new `index.d.ts` type declaration file, run:
+
+```sh
+npm install -g dts-gen
+dts-gen --module deepspeech --file index.d.ts
```
Comment (lissyx) [inline native_client/javascript/README.md:12]: It looks like that file needs to be packaged. Then `ts-node` seems more happy.

File: native_client/javascript/README.md (hunk 3 above)
Comment (reuben) [reply native_client/javascript/README.md:18]: My understanding is that `index.d.js` allows our package to support usage from both JavaScript and TypeScript, so having `index.d.js` means an `index.ts` is not needed.

**Comment by piraka9011** [comment]
Closing in favor of #2882

**Pull request closed without merging**